hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/TestHoodieSparkTable.java

```diff
@@ -0,0 +1,137 @@
+    when(writeMarkers.createdAndMergedDataPaths(getEngineContext(), writeConfig.getFinalizeWriteParallelism())).thenReturn(markerList);
+
+    HoodieTableMetaClient metaClient = mock(HoodieTableMetaClient.class);
+    when(metaClient.getBasePath()).thenReturn(new StoragePath(basePath));
+    when(metaClient.getTableType()).thenReturn(HoodieTableType.COPY_ON_WRITE);
+    HoodieStorage storage = mock(HoodieStorage.class);
+    when(metaClient.getStorage()).thenReturn(storage);
+
+    additionalFiles.forEach(fileName -> {
+      try {
+        StoragePath storagePath = new StoragePath(basePath + "/" + partitionPath + "/" + fileName);
+        if (failureType == DeleteFailureType.TRUE_ON_DELETE) {
+          when(storage.deleteFile(storagePath)).thenReturn(true);
+        } else if (failureType == DeleteFailureType.FALSE_ON_DELETE_IS_EXISTS_FALSE) {
+          when(storage.deleteFile(storagePath)).thenReturn(false);
+          when(storage.exists(storagePath)).thenReturn(false);
+        } else if (failureType == DeleteFailureType.FALSE_ON_DELETE_IS_EXISTS_TRUE) {
+          when(storage.deleteFile(storagePath)).thenReturn(false);
+          when(storage.exists(storagePath)).thenReturn(true);
+        } else if (failureType == DeleteFailureType.FILE_NOT_FOUND_EXC_ON_DELETE) {
+          when(storage.deleteFile(storagePath)).thenThrow(new FileNotFoundException("throwing file not found exception"));
+        } else {
+          // run time exception
+          when(storage.deleteFile(storagePath)).thenThrow(new RuntimeException("throwing run time exception"));
+        }
+        // lets create the data file. so that we can validate later.
+        localStorage.create(storagePath);
+      } catch (IOException e) {
+        throw new HoodieException("Failed to check data file existance " + fileName);
+      }
+    });
+    HoodieTable hoodieTable = HoodieSparkTable.create(writeConfig, getEngineContext(), metaClient);
+    if (failureType == DeleteFailureType.RUNTIME_EXC_ON_DELETE || failureType == DeleteFailureType.FALSE_ON_DELETE_IS_EXISTS_TRUE) {
+      assertThrows(HoodieException.class, () -> {
+        hoodieTable.reconcileAgainstMarkers(getEngineContext(), "0001", writeStatList, false, false, writeMarkers);
+      });
+    } else { // all other cases
+      hoodieTable.reconcileAgainstMarkers(getEngineContext(), "0001", writeStatList, false, false, writeMarkers);
+      // validate that additional files are deleted from storage
+      additionalFiles.forEach(fileName -> {
+        try {
+          verify(storage, times(1)).deleteFile(new StoragePath(basePath + "/" + partitionPath + "/" + fileName));
+        } catch (IOException e) {
+          throw new HoodieException("Failed to validate that file exists " + fileName);
+        }
+      });
+    }
+    System.out.println("Asdas");
+  }
+
+  enum DeleteFailureType {
+    TRUE_ON_DELETE,
+    FALSE_ON_DELETE_IS_EXISTS_FALSE,
+    FALSE_ON_DELETE_IS_EXISTS_TRUE,
```

Comment:
I think if `deletionSuccess` is false, then let's do a delete anyway as it is idempotent. No need to check of isExists.

Reply:
problem is, what incase next the subsequent delete also fails? thats why kept it simple. we can just check for exists and if it returns true, lets fail the call.